bridge/client/app/_components/ktb-deployment-list/ktb-deployment-list.component.ts

```diff
@@ -61,43 +60,56 @@ export class KtbDeploymentListComponent implements OnInit, OnDestroy {
   constructor(public _changeDetectorRef: ChangeDetectorRef, private route: ActivatedRoute, private dataService: DataService, private router: Router, private location: Location) { }
 
   ngOnInit(): void {
-    this.route.params
+    this.dataService._remediationsUpdated
       .pipe(takeUntil(this.unsubscribe$))
-      .subscribe(params => {
-        this.dataService.getProject(params.projectName)
-          .subscribe(project => {
-            this.projectName = project.projectName;
-            this.gitRemoteURI = project.gitRemoteURI;
-            this.service.deployments = project.getDeploymentsOfService(this.service.serviceName);
-            if (params.shkeptncontext && this.service.serviceName === params.serviceName) {
-              const paramDeployment = this.service.deployments.find(deployment => deployment.shkeptncontext === params.shkeptncontext);
-              if (paramDeployment) {
-                this.selectDeployment(paramDeployment, !params.stage);
-              } else {
-                const routeUrl = this.router.createUrlTree(['/project', this.projectName, 'service', params.serviceName]);
-                this.location.go(routeUrl.toString());
-              }
-            }
-            this.minPageSize = this.service.deployments.length;
-            this.updateDataSource();
-          });
-
-        this.dataService.roots
-          .pipe(takeUntil(this.unsubscribe$))
-          .subscribe(() => {
-            this._changeDetectorRef.markForCheck();
-          });
+      .subscribe(() => {
+        this._changeDetectorRef.markForCheck();
       });
+
+    const params$ =this.route.params
+      .pipe(takeUntil(this.unsubscribe$));
+
+    const project$ = params$.pipe(
```

Comment:
I think this subscription on project should work, since this just makes sure that the currently selected project is loaded. 

But, this component is missing a subscription to new root events and triggering a change detection, and probably the view is not polling for new root events?

Reply:
This component does not need any root events, because they are initially fetched when the details are opened.